tests/conftest.py

```diff
@@ -2314,3 +2315,57 @@ def fio_job_dict_fixture():
         """)
     job_dict = yaml.safe_load(template)
     return job_dict
+
+
+@pytest.fixture(scope='function')
+def pgsql_factory_fixture(request):
```

Comment:
This fixture may be used in other tests where it is important to check the status of pgsql pods during or after an admin/failure operation. It would be great if fixture supports that capability. Please check.

Reply:
When I use threads to run benchamark and return the object immediate after creation of benchmark the validation of pgbench is happening in teardown.
I will take this comment in next PR.

Reply:
LGTM